### ISSUE TITLE
tf.keras.layers.AveragePooling1D data_format init argument broken

### ISSUE THREAD

**Issue by aligirayhanozbay** [tensorflow/tensorflow#27295]
<em>Please make sure that this is a bug. As per our [GitHub Policy](https://github.com/tensorflow/tensorflow/blob/master/ISSUES.md), we only address code/doc bugs, performance issues, feature requests and build/installation issues on GitHub. tag:bug_template</em>

**System information**
- Have I written custom code (as opposed to using a stock example script provided in TensorFlow): Yes
- OS Platform and Distribution (e.g., Linux Ubuntu 16.04): Ubuntu GNU/Linux 18.04
- Mobile device (e.g. iPhone 8, Pixel 2, Samsung Galaxy) if the issue happens on mobile device: N/A
- TensorFlow installed from (source or binary): Binary
- TensorFlow version (use command below): 1.12
- Python version: 3.6.7
- Bazel version (if compiling from source): N/A
- GCC/Compiler version (if compiling from source): N/A
- CUDA/cuDNN version: CUDA 9.2, cuDNN 7.3.1
- GPU model and memory: Nvidia Titan V, P100, K80

You can collect some of this information using our environment capture [script](https://github.com/tensorflow/tensorflow/tree/master/tools/tf_env_collect.sh)
You can also obtain the TensorFlow version with
python -c "import tensorflow as tf; print(tf.GIT_VERSION, tf.VERSION)"

**Describe the current behavior**
When instantiating a tf.keras.layers.AveragePooling1D, the data_format argument does not properly change which axis over which the pooling is performed - the pooling operation is conducted across the 2nd dimension of the input tensor regardless of the init argument supplied.

To illustrate this, consider 2 layers instantiated as
```
pool_1 = tf.keras.layers.AveragePooling1D(data_format='channels_first', pool_size=64)(prev_layer0) 
pool_2 = tf.keras.layers.AveragePooling1D(data_format='channels_last', pool_size=64)(prev_layer1)
```

- Assuming the output shape of prev_layer0 is (None,32,None), pool_1 outputs (None,0,None) when it should be outputting (None,32,None)
- Assuming the output shape of prev_layer1 is (None,None,32), pool_2 outputs (None,None,32) correctly

I have confirmed this is NOT merely the model summary misreporting the output shape - if you run training or prediction with a model containing an output shape of (None,0,None) generated in this way, it will lead to a 'Floating point exception (core dumped)' error.

**Describe the expected behavior**
When the `data_format='channels_first'` init argument is supplied, AveragePooling1D layers should pool along the last dimension of the 3D input tensors instead of the penultimate dimension (which is the expected behaviour when `data_format='channels_last'` is supplied)

**Code to reproduce the issue**
Provide a reproducible test case that is the bare minimum necessary to generate the problem.
```
#Channels first case
input_0 = Input(shape=(1,None))
conv1d_1 = tf.keras.layers.Conv1D(filters=16, kernel_size=5, padding='same', activation=tf.nn.leaky_relu, data_format = 'channels_first')(input_0)
conv1d_2 = tf.keras.layers.Conv1D(filters=32, kernel_size=5, padding='same', activation=tf.nn.leaky_relu, data_format = 'channels_first')(conv1d_1)
pool_2 = tf.keras.layers.AveragePooling1D(data_format='channels_first', pool_size=64)(conv1d_2)
conv1d_3 = tf.keras.layers.Conv1D(filters=256, kernel_size=1, padding='same', activation=tf.nn.leaky_relu, data_format = 'channels_first')(pool_2)
mod = Model(input_0, conv1d_3)
mod.summary() #Note the output shape of pool_2 - it will be (None,0,None)
#print(mod.predict(tf.random.uniform((1,1,128), dtype = tf.keras.backend.floatx())).shape) #If you run this it will cause a core dump

#Channels last case
input_0 = Input(shape=(None,1)) #implement arbitrary input shapes later
conv1d_1 = tf.keras.layers.Conv1D(filters=16, kernel_size=5, padding='same', activation=tf.nn.leaky_relu, data_format = 'channels_last')(input_0)
conv1d_2 = tf.keras.layers.Conv1D(filters=32, kernel_size=5, padding='same', activation=tf.nn.leaky_relu, data_format = 'channels_last')(conv1d_1)
pool_2 = tf.keras.layers.AveragePooling1D(data_format='channels_last', pool_size=64)(conv1d_2)
conv1d_3 = tf.keras.layers.Conv1D(filters=256, kernel_size=1, padding='same', activation=tf.nn.leaky_relu, data_format = 'channels_last')(pool_2)

mod2 = Model(input_0, conv1d_3)
mod2.summary()

print(mod2.predict(tf.random.uniform((1,128,1), dtype = tf.keras.backend.floatx())).shape) #Outputs a tensor of shape (1,2,256) as expected
```

**Other info / logs**
Include any logs or source code that would be helpful to diagnose the problem. If including tracebacks, please include the full traceback. Large logs and files should be attached.

**Comment by jvishnuvardhan** [comment]
@aligirayhanozbay I cannot reproduce the issue with `!pip install tf-nightly-gpu-2.0-preview==2.0.0.dev20190724`. Please check the gist [here](https://colab.sandbox.google.com/gist/jvishnuvardhan/03c58eb73f4a965a40ecfcaf0671dccf/tf_27295_keras_pooling.ipynb). 

Looks like it was resolved. I am closing the issue but feel free to open the issue if it persists again. Thanks!
